pkg/hwinfo/hwinfo.go

```diff
@@ -1,15 +1,39 @@
 package hwinfo
 
 import (
-	"errors"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
 )
 
-var ErrNotPresent = errors.New("hwinfo: not present")
+type pciDeviceInfo struct {
+	Address     string `json:"address"`
+	DeviceType  string `json:"device_type,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	ProductName string `json:"product_name"`
+	Description string `json:"description,omitempty"`
+}
+
+type usbDeviceInfo struct {
+	Address     string `json:"address,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	DeviceID    string `json:"id"`
+	Description string `json:"description,omitempty"`
+}
+
+type monitorInfo struct {
+	ID          string `json:"id"`
+	Description string `json:"description,omitempty"`
+	VendorName  string `json:"vendor_name,omitempty"`
+	Size        string `json:"size,omitempty"`
+	Resolution  string `json:"resolution,omitempty"`
+}
 
 func Inventory() (map[string]interface{}, error) {
 	hw, err := fetchInventory()
 	if err != nil {
-		return nil, err
+		err = errors.Wrap(err, "hwinfo")
+		log.Error(err)
```

Comment:
Is it correct that we return _and_ log the error? Shouldn't the logging be done when the error is handled?

Reply:
Yes. We have already discussed this in some other PR before. The reason is the caller of this function (`handler.GetAllMeasurements()`) does not have any error logging code. This function now is similar to other functions called from GetAllMeasurements().

In this PR I just moved ALL logging from low-level implementation here, making it's more explicit and eliminating the code duplication.